pandas/core/tools/numeric.py

```diff
@@ -142,6 +144,12 @@ def to_numeric(arg, errors="raise", downcast=None):
     else:
         values = arg
 
+    if isinstance(values, NumericArray):
```

Comment:
can you add a comment here and below on why you need to do this (use the masks). its obvious with the diff, but just the code its not so much why you are doing this.

now we *might* actually want to make this a context manager to do this (but likley overkill for this one particular case).

Reply:
> can you add a comment here and below on why you need to do this (use the masks). its obvious with the diff, but just the code its not so much why you are doing this.

Done

> now we _might_ actually want to make this a context manager to do this (but likley overkill for this one particular case).

I'm down to do this (is follow-on ok or would you rather here?). I feel like it would be useful in a number of places (#37915 for example) and possibly some groupby algos

Reply:
Let's leave that for a separate PR

Reply:
Sounds good.

I may open an issue to discuss what we want the manager to look like before implementing

Reply:
yes separate PR